scripts/das_tool.sh

```diff
@@ -0,0 +1,130 @@
+
+HELP_USAGE
+}
+
+# Exit if command is called with no arguments
+validate_arguments "$#"
+
+# Parse command line arguments
+while (("$#")); do
+    case "$1" in
+        -h|--help ) usage; exit 0 ;;
+        -i )       input_dir=$(readlink -f "$2"); shift 2 ;;
+        -o )       out_dir=$(readlink -m "$2"); shift 2 ;;
+        -t )       threads="$2"; shift 2 ;;
+        -pt )      proteins=$(readlink -m "$2"); shift 2 ;;
+        -opts )    shift; das_opts="$@"; break ;;
+        * )        echo "Option '$1' not recognized"; exit 1 ;;
+    esac
+done
+
+# Verify that input directory exists
+validate_input_dir
+
+# Create output directory if it doesn't exists
+validate_output_dir
+
+# Activate conda environment
+activate_env metabiome-das_tool
+
+# Output info
+echo "Conda environment: $CONDA_DEFAULT_ENV"
+echo "Input directory: ${input_dir:?'Input directory not set'}"
+echo "Number of threads: ${threads:=1}"
+echo "$(DAS_Tool -v)"
+echo "DAS tool called with options: $das_opts"
+
+# Calculate bins with DAS Tool
+
+# if the predicted protein files are not provided
+if [[ ! -d "$proteins" ]]; then
+    for file  in "$input_dir"/*; do
+        if [[ "$file" == *.@(fa|fna|fasta) ]]; then
+            contig="$file"
+            core_name=$(get_core_name "$contig")
+            create_dir "$out_dir" "$core_name"
+
+
+            # Generate a comma-separated list of the scaffolds-to-bins tsv files
+            tsv_list=$(ls -1 "$input_dir"/*"$core_name"*.tsv | paste -d, -s)
+
+            # Run DAS_Tool
+            DAS_Tool --bins "$tsv_list" \
+                -c "$contig" \
+                -o "$out_dir"/"$core_name"/"$core_name" \
+                --threads "$threads" \
+                $das_opts
+
+
+        elif [[ ! "$file" == *.@(fa|fna|fasta) ]] && [[ ! "$file" == *.@(tsv) ]]; then
+            echo "$file is not either a bin or a scaffold-to-bin tsv file, so it will be ignore"
+        fi
+    done
+
+# if the predicted protein files are provided
+elif [[ -d "$proteins" ]]; then
+    for file  in "$input_dir"/*; do
+        if [[ "$file" == *.@(fa|fna|fasta) ]]; then
+            contig="$file"
+            core_name=$(get_core_name "$contig")
+            create_dir "$out_dir" "$core_name"
+
+            # Generate a comma-separated list of the scaffolds-to-bins tsv files
+            tsv_list=$(ls -1 "$input_dir"/*"$core_name"*.tsv | paste -d, -s)
+
+
+            # Prodigal fasta file
+            prodigal_file=$(echo "$proteins"/*"$core_name"*.faa)
+
+            # Check if the prodigal fasta file is correctly provided
+            if [[ -e "$prodigal_file" ]]; then
+
+                # Run DAS Tool
+                echo "Skipping Prodigal gene prediction"
+                DAS_Tool --bins "$tsv_list" \
+                    -c "$contig" \
+                    -o "$out_dir"/"$core_name"/"$core_name" \
+                    --threads "$threads" \
+                    --proteins "$prodigal_file" \
+                    $das_opts
```

Comment:
Is it possible to run das_tools without the `-opts` flag? I have a vague memory that writing
```
code \
$opts
```
instead of
```
code $opts
```
caused some problems, but I am not sure.

Reply:
Yes! You are right. In other wrapper scripts we have encountered that issue.  However, if you noticed in the command  from above https://github.com/Nesper94/Metabiome/pull/37#issue-643680710
 
> To run DAS Tool:
> 
> `metabiome-das_tool -i input -o das_tool_out -opts --search_engine diamond --write_bins 1`

It already has the -opts activated and it worked like a charm! But, if you considered that we must remove that `\`, I'll do it.

Reply:
Yes, I asked if it was possible to run it *without* the `-opts` flag because this is when the problems arise, not when the flag is used.

Reply:
ahh I understand. But it also worked, try it yourself:

First, download and decompress the usearch database which I attached to this comment ([usearch](https://github.com/Nesper94/Metabiome/files/6497345/usearch11.0.667_i86linux32.gz)). Then, just make this file executable (`chmod +x usearch11.0.667_i86linux32`). Before running DAS Tool,  you must have this database in the same directory where you are running DAS Tool command or have it in  your PATH. Then, just run metabiome-das_tool like so:

`metabiome-das_tool -i input -o das_tool_out `

Reply:
Hi! I got the following error:
```
mv: cannot stat '~/test/das_tool/das_tool_out/sample_human_gut/sample_human_gut_proteins.faa.scg': No such file or directory
single copy gene prediction using usearch failed. Aborting
```

However this seems not to be related with the `-opts` flag, so I think there is no need to change the script.

Reply:
Hi Juan! that's odd because I just re-ran this command and it worked. Can you show me what's inside `input/` directory and where did you place the `usearch` executable? For example, I placed the `usearch` in the directory `test` alongside the `das_tool` input directory. And from there, I just ran the above command. 

![usearch_loc](https://user-images.githubusercontent.com/57162811/119079055-f926ec80-b9bc-11eb-9424-80c1bde1b9a6.png)

> However this seems not to be related with the -opts flag, so I think there is no need to change the script.

Yep, I agree. Let's just wait if Stefany got the same error.
Thanks!

Reply:
Hi, I tried again and now it worked fine, I think the error I got was because I had not deleted the output directory from a previous run.